### DISCUSSION TITLE
I think I encountered a bug

### DISCUSSION THREAD

**Discussion by oshmoz** [ajeetdsouza/zoxide#193] [Q&A] [answered]
Hello,
While testing zoxide, I encountered the following error message: 'cd: The directory '-' does not exist' while trying to 'z -'. How can I find the source of this problem?
I'm testing on Ubuntu server 20.04 with fish shell and I installed zoxide 0.6.0 by copying the executable from the releases page. Other than that zoxide seems to work fine.

Thank you.

**Comment by ajeetdsouza** [comment, marked as answer]
Hey @oshmoz, this issue has already been reported [here](https://github.com/ajeetdsouza/zoxide/issues/178). I've been meaning to cut a release with the fix, but I haven't had the time. I'll try to do it this weekend :smile:

**Comment by oshmoz** [comment]
Thank you for your quick reply. So just a quick followup question, is there any repository that I can add to my apt command, so I will get updates automatically?

Thanks again for this great utility.

**Reply by ajeetdsouza** [reply]
There isn't an official APT repository at the moment. On Ubuntu 21.04 onwards, zoxide is available in the official repositories, but that would only be updated as and when the maintainers find the time. Your best bet would be to use the binaries from GitHub.

If you use a lot of Rust tools, it may make sense to install Rust and install your utilities directly from crates.io, using `cargo install`. You can update all of them at once by using [`cargo install-update -a`](https://github.com/nabijaczleweli/cargo-update). This is a fairly common way to do things - I install all my Python utilities using `pipx` rather than relying on official repositories, since they always lag behind.

If you'd like a single command to update everything, there is the [`topgrade`](https://github.com/r-darwish/topgrade) utility which updates software installed from different sources (I haven't used this myself, but it seems quite popular).

**Comment by oshmoz** [comment]
Great. Thanks again.

**Reply by oshmoz** [reply]
Just another quick newbie question. I'm using plain Ubuntu server 20.04 and on version 0.6 I downloaded "zoxide-x86_64-unknown-linux-gnu" and it worked. Now, on version 0.7 I can't seem to find this file. What am I missing? Can I just use the "musl" version?
Thanks.

**Reply by ajeetdsouza** [reply]
Yup, I've removed GNU releases because the musl ones run faster and are more portable.
